### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by RadoslavMonov**

### CODE REVIEW

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.*;
```
Comment (SavageStan) [inline src/test/java/com/monov/student/controller/StudentControllerIntTest.java]: Please do a Google research on how to turn off wild card imports (star imports) in IntelliJ Idea. Once turned off, go over all you classes and optimize imports in order to remove them.

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
```
Comment (SavageStan) [inline src/test/java/com/monov/student/controller/StudentControllerIntTest.java:26]: Need empty line between imports and start of the class/annotations

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+        webEnvironment = SpringBootTest.WebEnvironment.MOCK,
+        classes = StudentServiceApplication.class)
+@AutoConfigureMockMvc
+@TestPropertySource(properties = { "spring.config.location = classpath:integration-test-application.yaml" })
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (SavageStan) [inline src/test/java/com/monov/student/controller/StudentControllerIntTest.java]: Why do the tests have a fixed order? Test A creates data that is used by test B?

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        ItemIds ids = new ItemIds(Arrays.asList(1L,2L));
+        mvc.perform(post("/students/ids")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(asJsonString(ids)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(2)));
+    }
+
+    private static String asJsonString(final Object obj) {
```
Comment (SavageStan) [inline src/test/java/com/monov/student/controller/StudentControllerIntTest.java]: You can probably move this method to a test util class that is placed in the commons project.

File: src/test/resources/integration-test-application.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+server:
+  port: 9001
+
+spring:
+  application:
+    name: STUDENT-SERVICE
+  jpa:
+    defer-datasource-initialization: true
+
+eureka:
```
Comment (SavageStan) [inline src/test/resources/integration-test-application.yaml]: Are you sure we still need Eureka?

Comment (RadoslavMonov) [reply src/test/resources/integration-test-application.yaml]: Eureka removed

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java (hunk 4 above)
Comment (RadoslavMonov) [reply src/test/java/com/monov/student/controller/StudentControllerIntTest.java]: done

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java (hunk 3 above)
Comment (RadoslavMonov) [reply src/test/java/com/monov/student/controller/StudentControllerIntTest.java]: order removed. Data changes to DB are rolled back after each test

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java (hunk 2 above)
Comment (RadoslavMonov) [reply src/test/java/com/monov/student/controller/StudentControllerIntTest.java:26]: done

File: src/test/java/com/monov/student/controller/StudentControllerIntTest.java (hunk 1 above)
Comment (RadoslavMonov) [reply src/test/java/com/monov/student/controller/StudentControllerIntTest.java]: done

**Review by SavageStan — APPROVED** [review]

**Pull request merged**